org.eclipse.lsp4e/src/org/eclipse/lsp4e/LSPEclipseUtils.java

```diff
@@ -1132,4 +1136,32 @@ public static boolean hasCapability(Either<Boolean, ? extends Object> eitherCapa
 		}
 	}
 
+	/**
+	 * Execute Eclipse Search UI to search LSP references from the given document and
+	 * offset.
+	 *
+	 * @param document
+	 *            the document.
+	 * @param offset
+	 *            the offset.
+	 * @param display
+	 *            the display to use to execute the search.
+	 */
+	public static void searchLSPReferences(@NonNull IDocument document, int offset, Display display) {
+		LanguageServiceAccessor
+				.getLanguageServers(document,
+						capabilities -> hasCapability(capabilities.getReferencesProvider())) //
+				.thenAcceptAsync(languageServers -> {
+					if (languageServers.isEmpty()) {
+						return;
+					}
+					LanguageServer ls = languageServers.get(0);
```

Comment:
I realize you are just moving the code, but is the general pattern not to loop over all LSs, why take here just the first one?